File: 프로그래머스/unrated/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
class Solution {
    public String solution(String my_string, int[][] queries) {
        String answer = "";
        
        for(int i = 0; i < queries.length; i++) {
            StringBuilder sb = new StringBuilder();
            
            int start = queries[i][0];
            int end = queries[i][1];
            
            for(int j = 0; j < my_string.length(); j++) {
                int index;
                if(j >= start && j <= end) index = start + end - j;
                else index = j;
                sb.append(my_string.charAt(index));
            }
            
            my_string = sb.toString();
        }
        answer = my_string;
        return answer;
    }
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    let answer = my_string.split('');
    
    queries.forEach((item) => {
        let tmp = answer.slice(item[0], item[1] + 1).reverse();
        let spl = answer.splice(item[0], item[1] - item[0] + 1);
        
        tmp.forEach((add, idx) => {
            answer.splice(item[0] + idx, 0, add);
        });
    });
    
    answer = answer.join('');
    
    return answer;
}